Client/Ecosia/Extensions/HomepageViewController+Ecosia.swift

```diff
@@ -91,3 +99,16 @@ extension HomepageViewController: YourImpactDelegate {
         homePanelDelegate?.homePanel(didSelectURL: url, visitType: .link, isGoogleTopSite: false)
     }
 }
+
+extension HomepageViewController: NTPBookmarkNudgeViewDelegate {
+    func nudgeCellOpenBookmarks() {
+        homePanelDelegate?.homePanelDidRequestToOpenLibrary(panel: .bookmarks)
+        User.shared.hideBookmarksNTPNudgeCard()
```

Comment:
I'm sure there is some clear need to decorate the `User`'s `shared` instance with the ability to interact with bookmarks or cards. Asking as I need to become more familiar with the user's instance and its abilities overall 🙏 .
This comment is valid for all the following unfamiliar (to me) behaviors belonging to the user's shared instance.

Reply:
This was my initial assumption, that `User.shared` keeps this state, but now that you're bringing this up it would be good to get some confirmation on this. Basically I'm keeping different states because those different elements are shown at different points in time due to different criteria being required for them to show up.

Reply:
I guess that's the question regarding the `User.shared` @kimar @ecotopian

Reply:
Historically we have been using the `User.shared` struct to persist this kind of application state whether popups/banners should show or have been shown. So it's totally relatable that Marcus followed that path and I don't see an imminent issue with this. 

In the long run I can imagine that the User struct gets bloated and harder to maintain and oversee. If we agree that this is the case, it can make sense to split up/extract the data which is less user related and think of a different persistency mechanism. But this should not be part of Marcus' tasks here. Instead we can assess this dedicatedly.